src/components/Tool/Appeal/Modals/UpdateDonationsModal/DonationTable/DonationTable.tsx

```diff
@@ -94,21 +101,20 @@ export const DonationTable: React.FC<DonationTableProps> = ({
   });
 
   useEffect(() => {
-    if (!data?.donations.nodes.length) {
-      return;
+    if (preselectedDonations?.length) {
+      setSelectedDonations(preselectedDonations);
     }
-
-    const preselectedDonations = data.donations.nodes
-      .filter((donation) => donation.appeal?.id === appealId)
-      .map(createDonationRow);
-
-    setSelectedDonations(preselectedDonations);
-  }, [data]);
+  }, [preselectedDonations]);
 
   const nodes = data?.donations.nodes || [];
 
   const donations = useMemo(() => nodes.map(createDonationRow), [nodes]);
 
+  const isPreselectedDonation = (donation: DonationRow) =>
+    preselectedDonations.some((donationRow) => donationRow.id === donation.id);
+  const isDonationChecked = (donation: DonationRow) =>
+    selectedDonations.some((donationRow) => donationRow.id === donation.id);
```

Comment:
Same here

Reply:
Also please can you make it return a boolean? I know I coded it to be a `some`, but I think `find` would be better.

Reply:
> Also please can you make it return a boolean?

I'm not sure what you mean. `some` returns a boolean and since `some` short-circuits too, I don't see a benefit of using `find`. If you have another preference for `find` over `some` besides stopping iteration early, let me know.

Reply:
I'm sorry, I didn't know `some` short-circiuts. I know it returned a boolean, but I thought it ran through all the array.

Sorry for slowing this PR down.

Reply:
No worries, and thanks for the feedback!